packages/office-ui-fabric-react/src/components/Button/BaseButton.tsx

```diff
@@ -627,8 +629,12 @@ export class BaseButton extends BaseComponent<IBaseButtonProps, IBaseButtonState
   };
 
   private _onClick = (ev: React.MouseEvent<HTMLDivElement | HTMLAnchorElement | HTMLButtonElement>) => {
-    if (!this.props.disabled && this.props.onClick !== undefined) {
-      this.props.onClick(ev); // not cancelling event because it's not disabled
+    if (!this.props.disabled) {
```

Comment:
Same thing here, this could be `if (this.props.disabled){return;} <rest of the code/>`